convert.c

```diff
@@ -256,6 +257,117 @@ static int will_convert_lf_to_crlf(size_t len, struct text_stat *stats,
 
 }
 
+static struct encoding {
+	const char *name;
+	iconv_t *to_git;
+	iconv_t *to_worktree;
+	struct encoding *next;
+} *encoding, **encoding_tail;
+static const char *default_encoding = "utf-8";
+
+static int encode_to_git(const char *path, const char *src, size_t src_len,
+			 struct strbuf *buf, struct encoding *enc)
+{
+#ifndef NO_ICONV
+	char *dst, *re_src;
+	int dst_len, re_src_len;
+
+	if (!enc || (src && !src_len))
+		return 0;
+
+	/*
+	 * Looks like we got called from "would_convert_to_git()".
+	 * Since we have an encoding we must assume that we would encode
```

Comment:
This comment confused me a bit. Suggestion:

> There is no input string or buffer, but an encoding was specified. Apparently, something was intended to be encoded, so report this function call as invalid.

Reply:
How about:
```
	/*
	 * Looks like we got called from "would_convert_to_git()".
	 * This means Git wants to know if it would encode the content.
	 * Let's answer with "yes" since an encoding was specified.
	 */
```

Reply:
Okay, I get it now 😃. This comment is much better from my point of view!